src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -115,7 +115,93 @@ describe("FlashBanner", () => {
       />
     )
 
-    expect(wrapper.text()).toContain("Link expired. Resend verification email.")
+    expect(wrapper.text()).toContain("Link expired.")
+  })
+})
+
+describe("Email confirmation link expired", () => {
+  describe("user can request email confirmation (me?.canRequestEmailConfirmation=true)", () => {
+    it("user is prompted to re-request email confirmation if they can", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { canRequestEmailConfirmation: true },
+        },
+        queryString: "?flash_message=expired_token",
+      })
+
+      expect(wrapper.text()).toContain("Link expired.")
+    })
+
+    it.only("user seeing banner can click to re-trigger email confirmation message", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { canRequestEmailConfirmation: true },
+        },
+        queryString: "?flash_message=expired_token",
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              unconfirmedEmail: "ceo@blackwater.biz",
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Link expired.")
+
+      wrapper.find("button").first().prop("onClick")({} as any)
+      await flushPromiseQueue()
+      wrapper.update()
+
+      expect(wrapper.text()).toContain(
+        "An email has been sent to ceo@blackwater.biz"
+      )
+    })
+
+    it("user click to re-trigger is tracked", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { canRequestEmailConfirmation: true },
+        },
+        queryString: "?flash_message=expired_token",
+      })
+
+      expect(wrapper.text()).toContain("Link expired.")
+
+      wrapper.find("button").first().simulate("click")
+
+      expect(trackEvent).toHaveBeenCalledWith({
+        action_type: "Click",
+        subject: "Email Confirmation Link Expired",
+      })
+    })
+
+    it("user sees an error message if sendConfirmationEmail mutation fails", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { canRequestEmailConfirmation: true },
+        },
+        queryString: "?flash_message=expired_token",
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              mutationError: {
+                error: "BadError",
+                message: "Something Bad",
+              },
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Link expired.")
+
+      wrapper.find("button").first().prop("onClick")({} as any)
```

Comment:
simulate("click")

Reply:
Oh interesting - I wasn't aware of this! Thank you for clarifying – I read the issue and looks like there's a fair bit of back and forth on the semantics of shallow rendering vs. full DOM rendering with respect to the simulate API.

Regardless, I think, on principle, being consistent is more important than preparing for this API to be deprecated. A find+replace across the codebase when the API does indeed become deprecated should be pretty straightfoward.